src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,218 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.Timer;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+/**
+ * Singleton that represents the shooter mechanism.
+ */
+public class Shooter extends Subsystem {
+
+    // TODO: Integrate with other subsystems for real
+    // TEMPORARY STUFF BEGINS HERE
+    private static final int ROLLER_PORT = 0;
+    private static final int MAX_CAPACITY = 5;
+
+    // TODO: Tune these values
+    private static final double ROLLER_SPEED = 0.5;
+    private static final double SPINUP_DELAY_SECONDS = 0.5;
+    private static final double FIRE_DURATION_SECONDS = 0.5;
+
+    private static class TurretPosition {
+        private double m_azimuth, m_elevation;
+
+        public TurretPosition(double azimuth, double elevation) {
+            m_azimuth = azimuth;
+            m_elevation = elevation;
+        }
+
+        public double getAzimuth() { return m_azimuth; }
+        public double getElevation() { return m_elevation; }
+    }
+
+    @FunctionalInterface
+    private interface Turret {
+        void set(TurretPosition position);
+    }
+
+    @FunctionalInterface
+    private interface Vision {
+        TurretPosition calcTargetPosition();
+    }
+
+    private interface Storage {
+        boolean isBallStopOpen();
+        boolean isFull();
+        void engageBallStop();
+        void releaseBallStop();
+    }
+
+    // TEMPORARY STUFF ENDS HERE
+
+    // Aggregation
+    private static Shooter instance;
+    private WPI_TalonSRX m_roller;
+    private Turret m_turret;
+    private Vision m_vision;
+    private Storage m_storage;
+
+
+    // State variables
+    private boolean m_ready = false;
+    private boolean m_spinningUp = false;
+    private boolean m_firing = false;
+    private int m_buffer = 0;
+    private Timer m_spinUpTimer;
+    private Timer m_fireTimer;
+
+    public static Shooter getInstance() {
+        if (instance == null)
+            instance = new Shooter();
+        return instance;
+    }
+
+    private Shooter() {
+        m_roller = new WPI_TalonSRX(ROLLER_PORT);
+
+        // TODO: Replace these with real subsystems
+        m_turret = position -> System.out.println(
+            "Setting dummy turret position to ("
+            + position.getAzimuth()
+            + ", "
+            + position.getElevation()
+            + ")"
+        );
+        m_vision = () -> {
+            System.out.println("Getting dummy vision target");
+            return new TurretPosition(0, 0);
+        };
+        m_storage = new Storage() {
+            private boolean open = false;
```

Comment:
m_open?

Reply:
got rid of this one, wasn't used anywhere